test/SimpleNameRegisterTest.t.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "ds-test/test.sol";
+import 'src/SimpleNameRegister.sol';
+
+interface CheatCodes {
+    function prank(address) external;
+    function expectRevert(bytes calldata) external;
+    function startPrank(address) external;
+    function stopPrank() external;
+}
+
+abstract contract StateZero is DSTest {
+    SimpleNameRegister public simpleNameRegister;
+    CheatCodes cheats;
+    
+    function setUp() public virtual {
+        simpleNameRegister = new SimpleNameRegister();
+        cheats = CheatCodes(HEVM_ADDRESS);
+    }
+}
+
+contract StateZeroTest is StateZero {
+
+    function testCannotRelease(string memory testString) public {  
+        cheats.expectRevert(bytes("Not your name!"));
+        simpleNameRegister.release(testString);
```

Comment:
it's not terrible to use this testing contract as the "user" but if it were a larger test suite it could get confusing.  just as us set up adversary in the registered state, its probably better to set up a primary user that can be used in the zero state.

also, if you use the `label` cheat command, the stack traces will show the string and are easier to follow

Reply:
Oh this is really useful in making sense of traces!